### ISSUE TITLE
Can I generate FormlyFieldConfig

### ISSUE THREAD

**Issue by stefaneicher** [ngx-formly/ngx-formly#3966]
I have a Java backend and Java has the lead. It would be nice to write server-side code once and generate the FormlyFieldConfig.
I there a way to do this or has my idea big flaws is have not yet understood?

This would make me write most to the logig in the backend and I have not to update change always in the frontend and backend.
I try to think about this for an application with a lot of forms. Maintaining them in backend and frontend will be a maintenence burden that I would like to avoid.
Regards Stefan

EmailPasswordForm.java:
package com.example.demo;

import lombok.Data;

@Data
public class EmailPasswordFrom {
    private String email;
    private String password;
}

[
  {
    "key": "email",
    "type": "input",
    "templateOptions": {
      "label": "Email",
      "placeholder": "Enter your email",
      "required": true
    }
  },
  {
    "key": "password",
    "type": "input",
    "templateOptions": {
      "type": "password",
      "label": "Password",
      "placeholder": "Enter your password",
      "required": true
    }
  }
]

**Comment by skpanagiotis** [comment]
In my opinion generating FormlyFieldConfig Interface/Model in backend is a **false approach**. It's much-much better your backend returns a simpler json for example something like:

```json
[
  {
    "key": "email",
    "type": "input",
    "fieldType": "email"
    "label": "Email"
    "placeholder": "Enter your email",
    "mandatory": true,
  },
  {
    "key": "password",
    "type": "input",
    "fieldType": "password"
    "label": "Password"
    "placeholder": "Enter your password",
    "mandatory": true,
  },
]
```

And then your front end takes this json and mapping it into a **FormlyFieldConfig** object. I think this is the best approach when the form fields and the form in general are created by Backend.
In addition think that Frontend may needs some validators for some fields or some methods which are called by fields or even some custom styling. It will be very difficult or even impossible to handle these requirements from the Backend. 

To summarize the best way and the most popular approach is the Backend returning a custom Object that includes some basic properties for each form field, and then you Frontend will transform this custom object into FormlyFieldConfig, with the neccesarry styling, validation, methods, wrappers etc.
